### ISSUE TITLE
Iris Classification using DL

### ISSUE THREAD

**Issue by Sindhu-2004** [abhisheks008/DL-Simplified#598]
The goal of this task is to create a model that can accurately predict the species of an iris flower based on its sepal length, sepal width, petal length, and petal width.

**Comment by abhisheks008** [comment]
Can you share the dataset and approach for solving this issue?

**Comment by Sindhu-2004** [comment]
Project Title : Iris Flower Classification
Aim : The goal of this task is to create a model that can accurately predict the species of an iris flower based on its sepal length, sepal width, petal length, and petal width.
Dataset : 
[Iris Flower Dataset] https://www.kaggle.com/datasets/arshid/iris-flower-dataset
Approach : Logistic Regression, Classification, scikit-learn, TensorFlow, and PyTorch.

**Comment by abhisheks008** [comment]
One issue at a time.

**Comment by Mehaksaini2808** [comment]
Hey! If not assigned to anyone I would love to work on this
Project Title: Iris Flower Classification
Aim: The goal of this task is to create a ML model that aims to predict the species of an Iris flower based on its morphological features.
Dataset :
https://www.kaggle.com/datasets/uciml/iris
Approach : 
- Performing EDA to help understand the dataset and to identify patterns, relationships, and potential issues like outliers or missing values.
-  Data Processing - To handle missing values, feature scaling, encode categorical data
-  Model - Training with the help of Logistic Regression, SVM, and Random forest
- Model Evaluation -  Accuracy, Confusion matrix, precision, recall f1 score and ROC-AUC curve.

**Comment by Mehaksaini2808** [comment]
1. Full name : Mehak Saini
2. Email ID : mehaksaini288@gmail.com

What is your participant role? (Mention the Open Source program)Contributor in GSSOC'24

**Comment by Mehaksaini2808** [comment]
Please assign this issue to me.

**Comment by abhisheks008** [comment]
> 1. Full name : Mehak Saini
> 2. Email ID : [mehaksaini288@gmail.com](mailto:mehaksaini288@gmail.com)
> 3. Dataset : https://www.kaggle.com/datasets/uciml/iris
> 4. Approach : Logistic Regression, SVM and random forest. Perform model Evaluation and select the best performing model.
>    What is your participant role? (Mention the Open Source program)Contributor in GSSOC'24

This repo majorly focuses on deep learning methods rather than machine learning techniques.

**Comment by Mehaksaini2808** [comment]
Sorry for the confusion!

I can implement 3 deep learning model for this dataset and find the best-fitted one by comparing them all based on their accuracy.

**Comment by abhisheks008** [comment]
This issue is raised by a contributor for personal contribution, I don't think you can work on this issue. 

You can check out the issues raised by me and having `Up for Grabs` labels.

**Comment by wildcraft958** [comment]
Full name : Animesh Raj
Dataset : https://www.kaggle.com/datasets/uciml/iris
Techstack: pandas, matplotlib, seaborn, pytorch
Approach : 1. Data cleaning
2. EDA
3. Models i will be using are FNN and CNN.
4. Evaluation
What is your participant role? KWOC'24

**Comment by SupratikB23** [comment]
Hello, I would love to work on this project under SWOC'25.

**Project Title:** Iris Flower Classification
**Name:** Supratik Bhowal
**Participant role:** SWOC'25
**Dataset:** https://www.kaggle.com/datasets/arshid/iris-flower-dataset

**Approach:**
1. Models Used:
     - Use MLP (FNN)
     - Use Deep NN applying Dropout for regularization
     - Use CNN having 2D convolutional layers (architechture: LeNet-5 -> for simplicity)
 2. All Model Compilation will be done using categorical cross-entropy as the loss func and also will use the Adam optimizer
 3. Scatterplot Matrix, Confusion Matrix Heatmap, Distribution Plots along with Training and Validation Accuracy & Loss for each model used.

**Comment by abhisheks008** [comment]
> Hello, I would love to work on this project under SWOC'25.
> 
> **Project Title:** Iris Flower Classification **Name:** Supratik Bhowal **Participant role:** SWOC'25 **Dataset:** https://www.kaggle.com/datasets/arshid/iris-flower-dataset
> 
> **Approach:**
> 
> 1. Models Used:
>    
>    * Use MLP (FNN)
>    * Use Deep NN applying Dropout for regularization
>    * Use CNN having 2D convolutional layers (architechture: LeNet-5 -> for simplicity)
> 2. All Model Compilation will be done using categorical cross-entropy as the loss func and also will use the Adam optimizer
> 3. Scatterplot Matrix, Confusion Matrix Heatmap, Distribution Plots along with Training and Validation Accuracy & Loss for each model used.

It's iris classification, not the flower one :(

**Comment by SupratikB23** [comment]
> for

So, is it this dataset: 
↓
https://www.kaggle.com/datasets/naureenmohammad/mmu-iris-dataset

Or others like finding Iris diseases?

**Comment by Adhavan1801** [comment]
Hello @abhisheks008,

I would like to work on this issue.

Before proceeding, I would like to confirm the exact dataset/task since there was some confusion in the discussion regarding flower classification vs iris-based image classification.

If assigned, I plan to:

* perform preprocessing and exploratory analysis
* implement deep learning models suitable for the dataset
* compare architectures like MLP/FNN and lightweight CNNs (if image-based)
* evaluate using accuracy, confusion matrix, and loss curves
* provide clean documentation and reproducible code

Participant Role: Contributor

Looking forward to contributing to this repository.

**Comment by abhisheks008** [comment]
This will be Iris Based Image classification.

Can you share the final dataset and the models you are going to implement?

**Comment by Adhavan1801** [comment]
Dataset:
I'll be using the Iris Flower Image Dataset from Kaggle:
🔗 https://www.kaggle.com/datasets/jeffheaton/iris-computer-vision

Models to be implemented (3 models):
1 . Custom CNN  -Built from scratch — baseline model
2 . MobileNetV2  -Transfer Learning — lightweight & efficient
3 . EfficientNetB0 - Transfer Learning — best accuracy/compute r

**Comment by Dipti-Makwana** [comment]
Hi @abhisheks008! I am a GSSoC 2026 selected contributor on the Open Source + AI Agents track. I would love to work on this Iris Image Classification issue.
Dataset: Iris Computer Vision Dataset from Kaggle
🔗 https://www.kaggle.com/datasets/jeffheaton/iris-computer-vision
Models I plan to implement (3 models):

Custom CNN — built from scratch as baseline model
MobileNetV2 — Transfer Learning for efficiency
ResNet50 — Transfer Learning for best accuracy

Approach:

Data preprocessing and augmentation
EDA with visualizations
Train and compare all 3 models
Evaluate using accuracy, confusion matrix, and loss curves
Clean Jupyter notebook with proper README.md

Please assign this issue to me. Thank you! 🙏
